### PR TITLE
Flutter SDK: tests for services

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: templates/flutter/test/services/service_test.dart.twig [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                headers: argThat(isNotNull),
+            )).thenAnswer((_) async => Response(data: data));
+            {%~ else ~%}
+            when(client.call(
+                HttpMethod.{{method.method | caseLower}},
+            )).thenAnswer((_) async => Response(data: data));
+            {%~ endif ~%}
+
+            final response = await {{service.name | caseCamel}}.{{method.name | caseCamel}}({%~ for parameter in method.parameters.all | filter((param) => param.required) ~%}
+                {{parameter.name | escapeKeyword | caseCamel}}: {% if parameter.type == 'object' %}{}{% elseif parameter.type == 'file' %}InputFile.fromPath(path: './image.png'){% else %}'{{parameter.example}}'{%~ endif ~%},{%~ endfor ~%}
```
Comment (stnguyen90) [inline templates/flutter/test/services/service_test.dart.twig:95]: This needs to include a `{% elseif parameter.type == 'array' %}[]` or else the teams_test.dart will fail:

image

```suggestion
                {{parameter.name | escapeKeyword | caseCamel}}: {% if parameter.type == 'object' %}{}{% elseif parameter.type == 'array' %}[]{% elseif parameter.type == 'file' %}InputFile.fromPath(path: './image.png'){% else %}'{{parameter.example}}'{%~ endif ~%},{%~ endfor ~%}
```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

**Comment by stnguyen90** [comment]
Also, mockito should be added to the pubspec.yaml, right?

**Comment by lohanidamodar** [comment]
> Also, mockito should be added to the pubspec.yaml, right?

should be added, I must have added, if not feel free to add

File: templates/dart/test/src/exception_test.dart.twig [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import 'package:{{lang.params.packageName}}/src/exception.dart';
```
Comment (stnguyen90) [inline templates/dart/test/src/exception_test.dart.twig:1]:
```suggestion
import 'package:{{ language.params.packageName }}/src/exception.dart';
```

File: templates/dart/test/src/exception_test.dart.twig [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import 'package:flutter_test/flutter_test.dart';
```
Comment (stnguyen90) [inline templates/dart/test/src/exception_test.dart.twig:2]: To make this work in dart and flutter this should be:

```suggestion
{% if 'dart' in language.params.packageName %}
import 'package:test/test.dart';
{% else %}
import 'package:flutter_test/flutter_test.dart';
{% endif %}
```

**Review by stnguyen90 — APPROVED** [review]

**Pull request merged**